### ISSUE TITLE
[HOLD for payment 2024-02-26] [$500] App does not focus to first option on search if selected from first page option in timezone

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29080]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Open settings->profile->timezone
3. Open timezone and scroll to top
4. Select any one timezone from the visible timezones when scrolled to top eg: Africa/Asmara
5. Search any term in search (eg: IN) and observe that focus is not changed to first value as it normally does for all other similar UI

## Expected Result:
App should shift focus to first option on search in timezone

## Actual Result:
After scroll to top and selecting any one timezone from visible options, app does not shift focus to first option on search in timezone

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.79.3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/a9ee2e41-187c-4095-980e-97af702999dd

https://github.com/Expensify/App/assets/93399543/fd932312-3d29-4096-9c28-ce662ea5169a

https://github.com/Expensify/App/assets/93399543/6b6311c0-389d-4b98-b696-c1cc716f35a0

https://github.com/Expensify/App/assets/93399543/fbd86a63-4943-48fc-ab51-9f17acd20268

https://github.com/Expensify/App/assets/93399543/0bf66c86-a9c0-437a-94f1-b3dbc146db5c

https://github.com/Expensify/App/assets/93399543/ceff9bce-a5d3-4666-a29d-d3c42eb412b4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1696672477964269

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0178208b9c15b89907</li>
        <li>Upwork Job ID: 1711342124473376768</li>
        <li>Last Price Increase: 2023-10-16</li>
<li>Automatic offers: </li>
<ul>
<li>abzokhattab | Contributor | 27368232</li>
<li>dhanashree-sawant | Reporter | 27368237</li>
</ul></li>
    </ul>
</details>

**Comment by Santhosh-Sellavel** [comment]
@jliexpensify Can you reassign a new C+ by applying `external` label again? Unassigning this one due to low bandwidth

**Comment by maxconnectAbhi** [comment]
Not able to replicate

**Comment by jliexpensify** [comment]
Thanks @maxconnectAbhi - looks like the app is v80-1, so I will also confirm

**Comment by jliexpensify** [comment]
Hmm it still seems to be an issue on the Android app - when Is search `In`, my assumption is that it would default me to the country first (i.e. India). Instead, it shows me `America/Indiana` - so I don't think this is fixed.

**Comment by robertKozik** [comment]
"@jliexpensify, is the focus applied on the first one option though? Correct me if I'm wrong, but from your comment it looks like a different issue connected with the search function."

**Comment by jliexpensify** [comment]
Hmm maybe? I think this is what I'm unclear on:

>observe that focus is not changed to first value as it normally does for all other similar UI

Does this mean that when typing `IN`, then whatever is focused on should be bolded?

**Comment by robertKozik** [comment]
I came back to it, and I can confirm that this bug is still reproductible. I'll review the proposal

**Comment by robertKozik** [comment]
@abzokhattab, I appreciate your proposal. Since you already have the proof of concept implemented, could you verify if it behaves as intended for the case shown in the video? Specifically, select Africa/Asmara and then type in. I've found it to be quite dependent on user input, so I'd appreciate a double-check on a scenario that is currently not working.

**Comment by abzokhattab** [comment]
Hi @robertKozik  yeah i already used `Africa/Asmara` in the attached POC above please check it again. 

let me know if you want to test it with any other values

**Comment by MonilBhavsar** [comment]
@abzokhattab sorry could you please explain root cause in other way - why we're not focussing the first option like we do in OptionSelector. I see we're using SelectionList component at many places. So is focussing not working as expected in that component?

**Comment by abzokhattab** [comment]
@MonilBhavsar 

>  I see we're using `SelectionList` component at many places. So is focussing not working as expected in that component? 

Yes focusing on the first item when the options list changes is not a feature in the `SelectionList` component so this issue exists in all components that uses `SelectionList`, here is another example:  the workspace/settings/currency page which uses the `SelectionList` component: 


https://github.com/Expensify/App/assets/59809993/ec1bfbb9-3af8-4398-ae51-5b79a78362e5

If we want the focus to be on the first item whenever the `sections` list changes like in the `OptionsList`,

we can acheive that by adding this useEffect:

```js
    useEffect(() => {
        //do not change focus on the first render, as it should focus on the selected item
        if (firstLayoutRef.current) {
            return;
        }
        // set focus to the first item when the sections list is changed
        setFocusedIndex(0);
    }, [sections]);

```


Result: 

https://github.com/Expensify/App/assets/59809993/06fbb3a1-af83-4b64-9b4f-ba557e883ef8


Will update the proposal as well.

**Comment by MonilBhavsar** [comment]
Thanks! I would vote to fix focussing on SelectionList component rather than using OptionListSelector instead of SelectionList. @robertKozik could you please take a look at the updated proposal

**Comment by MonilBhavsar** [comment]
Awaiting review of updated proposal from @robertKozik

**Comment by robertKozik** [comment]
Sorry for the delay 🙇🏼 The updated proposal looks good as well, If we don't want to replace the component. Let's see it in action to test all the cases.

CC. @MonilBhavsar

**Comment by MonilBhavsar** [comment]
Agree. We should test all pages that uses SelectionList component

**Comment by abzokhattab** [comment]
PR is ready

**Comment by situchan** [comment]
@abzokhattab can you please handle [this regression](https://github.com/Expensify/App/pull/30438#issuecomment-1789081358)? It's blocking another issue.

**Comment by abzokhattab** [comment]
Looking into it today

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App focuses on first item when refreshing the workspace/currency page


### What is the root cause of that problem?

the `sections` gets rerendered multiple times caussing the useEffect in this proposal to focus on the first item on refresh the selection page https://github.com/Expensify/App/issues/29080#issuecomment-1752827570

### What changes do you think we should make in order to solve the problem?

instead of using the `useEffect`, it makes more sense to change the focus to the first item on changing the "input" not when the sections change because it has some edge cases. 

so I would remove the changes we made in the first proposal and change it to: 


```js
    const onChangeTextHelper = useCallback(
        (value) => {
            onChangeText(value);
            updateAndScrollToFocusedIndex(0);
        },
        [onChangeText, updateAndScrollToFocusedIndex],
    );
```
 

### POC 


https://github.com/Expensify/App/assets/59809993/e8d49139-8808-4d47-8415-79a6240e5dbe

**Comment by MonilBhavsar** [comment]
@robertKozik would like to have your thoughts here

**Comment by MonilBhavsar** [comment]
Removed Awaiting payment labels and from title since we need to fix this before issuing payment

**Comment by DylanDylann** [comment]
regression from this PR https://github.com/Expensify/App/pull/30438

**Comment by jliexpensify** [comment]
Bump @robertKozik

**Comment by wlegolas** [comment]
Hello everyone.

I sent this proposal https://github.com/Expensify/App/issues/31087#issuecomment-1803039218 that is related to the same problem.

**Comment by robertKozik** [comment]
Sorry for the delay here, I believe changing the implementation and unifying it with the battle-tested approach from `OptionsSelector` (described by @tienifr [here](https://github.com/Expensify/App/issues/29080#issuecomment-1797811942)) is the way to go here. More tweaking of the current approach can lead to even more regressions/problems, while the `OptionSelector` way to achieve this is currently working as expected

**Comment by abzokhattab** [comment]
I think it makes more sense to be on input change though @robertKozik  .. since the requirement is "the first item is selected whenever the user changes the search value in the text input field"

**Comment by jliexpensify** [comment]
Going to bump @MonilBhavsar so he doesn't miss [this](https://github.com/Expensify/App/issues/29080#issuecomment-1805227607) - he'll have an answer for you!

**Comment by s-alves10** [comment]
@robertKozik @MonilBhavsar 

This is a general issue for `SelectionList`.
I think we can simply solve both https://github.com/Expensify/App/issues/31110 and this issue by replacing the dependency of `useEffect` with `textInputValue`. This makes sense because the purpose is to focus the first item when search input is changed and `textInputValue` is synchronized with `sections`.
Please check [this proposal](https://github.com/Expensify/App/issues/31110#issuecomment-1803777378) as well.

**Comment by MonilBhavsar** [comment]
@tienifr's propsal to align with OptionsSelector behavior looks good. Could you please confirm if your proposal also fixes the regression mentioned here https://github.com/Expensify/App/pull/30438#issuecomment-1789081358
---

@robertKozik could you please take a look at this proposal - https://github.com/Expensify/App/issues/29080#issuecomment-1812046352

**Comment by sourcecodedeveloper** [comment]
@robertKozik since proposals are still being reviewed, please also test [proposal](https://github.com/Expensify/App/issues/31087#issuecomment-1803506915), The proposal resolves the regression https://github.com/Expensify/App/pull/30438#issuecomment-1789081358, https://github.com/Expensify/App/issues/31087 and https://github.com/Expensify/App/issues/31110.

**Comment by MonilBhavsar** [comment]
Waiting for proposals to be reviewed

**Comment by jliexpensify** [comment]
@robertKozik can you please review the proposals? It looks like we are blocked on you - if you don't respond in the next 24 hours, I'm going to reassign this to another C+ cc @MonilBhavsar (unless you think there's a reason not to?)

**Comment by jliexpensify** [comment]
@fedirjh Hi, could you help us push this issue along? We haven't heard from Robert in over 2 weeks so I thought it would be best to reassign.

Thanks!

**Comment by s-alves10** [comment]
@fedirjh 

Did you check https://github.com/Expensify/App/issues/29080#issuecomment-1812046352?

**Comment by sourcecodedeveloper** [comment]
@fedirjh Both the timezone selection page and the default currency page use the selection list. If the [PR](https://github.com/Expensify/App/pull/30438) would have caused the regression then that would have also affected the timezone selection page. The issue https://github.com/Expensify/App/issues/29080 should be closed and https://github.com/Expensify/App/pull/30438#issuecomment-1789081358 should not be treated as regression but handled as separate issue https://github.com/Expensify/App/issues/30951.

**Comment by s-alves10** [comment]
@fedirjh 

Let me explain why I suggest the use of `useEffect`

When search input changes, `onChangeText` is called. `onChangeText` calls setter function of `useState` as you can see below
https://github.com/Expensify/App/blob/48e0c1e435ae98df6d8ad2dda5caf624e73a0792/src/pages/workspace/WorkspaceSettingsCurrencyPage.js#L97

In the next rendering cycle, `searchText` will change and `sections` would change as well
https://github.com/Expensify/App/blob/48e0c1e435ae98df6d8ad2dda5caf624e73a0792/src/pages/workspace/WorkspaceSettingsCurrencyPage.js#L67

In `SelectionList` component, `flattenedSections` and `updateAndScrollToFocusedIndex` change as well in the same rendering cycle
https://github.com/Expensify/App/blob/48e0c1e435ae98df6d8ad2dda5caf624e73a0792/src/components/SelectionList/BaseSelectionList.js#L86

In my solution, I use the `textInputValue` as the dependency of `useEffect`. So the effect callback is called right after the updated list is rendered onto the screen.

Comparing the solution suggested by @abzokhattab, his solution is calling the `updateAndScrollToFocusedIndex` inside the `onChangeText` callback. This means `updateAndScrollToFocusedIndex` is called before it's updated by the updated `flattenedSections` (`flattenedSections` is the dependency of the function). 

As explained above, what I wanted to do is to scroll the list right after the list is updated. I think this makes more sense than scrolling the list before the list is updated

Please let me know your thoughts
Thanks

**Comment by sourcecodedeveloper** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Open workspace currency settings pages and refresh website. The selected currency is no longer highlighted and the list scroll to the top.

### What is the root cause of that problem?
The workspaceSettingsCurrencyPage is using withPolicyAndFullscreenLoading which reloads data on page refresh.
https://github.com/Expensify/App/blob/48e0c1e435ae98df6d8ad2dda5caf624e73a0792/src/pages/workspace/WorkspaceSettingsCurrencyPage.js#L113

The firstLayoutRef returns true for the first render and skips setting the focus index but due to withPolicyAndFullscreenLoading the firstLayoutRef returns false on subsequent calls and the updateAndScrollToFocusedIndex(0) is called causing the list to scroll to top.

https://github.com/Expensify/App/blob/48e0c1e435ae98df6d8ad2dda5caf624e73a0792/src/components/SelectionList/BaseSelectionList.js#L360


### What changes do you think we should make in order to solve the problem?
Listening to section changes or flattenedSections skips the first render but due to withPolicyAndFullscreenLoading subsequent render result in firstLayoutRef return false and the updateAndScrollToFocusedIndex(0) causing the list to scroll to the top. The useEffect should listen to textInputValue, also need to handle if no search text is entered and there is no selection then the first member of the list should not be higlighted. If the textInputValue is not empty then set focus index to 0 and scroll to the top otherwise scroll to the selectedOption.


    useEffect(() => {
        // do not change focus on the first render, as it should focus on the selected item
        if (firstLayoutRef.current) {
            return;
        }

        // set the focus on the first item when the sections list is changed
        if (sections.length > 0) {
            const selectedOptions = flattenedSections.selectedOptions;
            if (!canSelectMultiple) {
                if(!textInputValue && _.isEmpty(selectedOptions))
                {
                    // textInputValue is empty, set focus to -1 to avoid showing the highlight on the first member.
                    setFocusedIndex(-1);
                    return;
                }
                    // textInputValue is non-empty, set the focus index to 0 to focus on first item in list
                if (textInputValue || _.isEmpty(selectedOptions)) {
                    updateAndScrollToFocusedIndex(0);
                }
                else {
                    // when textInputValue is non-empty, focusedIndex is set to 0, need to get the index of selectedOption.
                    updateAndScrollToFocusedIndex(selectedOptions[0].index);
                }
            }
            else {
                // For Multiple options selectRow is handling focus.
            }
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [textInputValue]);

Need to capture index of selectedOption because when filtering list and setting focus to first value resets the focusedIndex.
https://github.com/Expensify/App/blob/48e0c1e435ae98df6d8ad2dda5caf624e73a0792/src/components/SelectionList/BaseSelectionList.js#L122

    selectedOptions.push({
        ...item,
        index: optionIndex
    });

### What alternative solutions did you explore? (Optional)

**Comment by sourcecodedeveloper** [comment]
@fedirjh Please review proposal. Calling updateAndScrollToFocusedIndex(0) inside onChangeText and then clearing search filter results in the first item being focused instead of the selectedOption.
Also in invite members page when filtering and clearing the search text, the first member is highlighted when there should be no highlight. When using sections or flattenedSections in useEffect, because the WorkspaceSettingsCurrencyPage is using withPolicyAndFullscreenLoading, the first render is skipped by firstLayoutRef but for subsequent re-renders firstLayoutRef returns false and then the updateAndScrollToFocusedIndex(0) scrolls the list to the top.

**Comment by jliexpensify** [comment]
Have bumped Fedi in Slack

**Comment by s-alves10** [comment]
@fedirjh 

> could you clarify the rationale behind delaying the update of the focusedIndex until the next rendering cycle instead of updating it within the current cycle? It make sense to defer the update to the next rendering cycle only if the focused index is derived from the flattenedSections.

As you mentioned, we're handling focused index and so use-callback based solution works fine as well. But logically, I think the intention is to focus the first item of the updated list. There may exist several solutions for an issue, but the ideal solution should be logically clean in my opinion. This is just my thought. Thanks

**Comment by sourcecodedeveloper** [comment]
> @s-alves10 Thank you for providing comprehensive feedback. However, could you clarify the rationale behind delaying the update of the `focusedIndex` until the next rendering cycle instead of updating it within the current cycle? For your information, presently, we consistently update to the zero index whenever the text undergoes changes. It make sense to defer the update to the next rendering cycle only if the focused index is derived from the `flattenedSections`.
> 
> > Calling updateAndScrollToFocusedIndex(0) inside onChangeText and then clearing search filter results in the first item being focused instead of the selectedOption.
> 
> @sourcecodedeveloper I believe this is the expected result across the App.
> CleanShot.2023-12-05.at.01.52.18.mp4

@fedirjh when the search filter is cleared and if there is selected item available for e.g. in the timezone or default currency list then that selected item needs to be focused and not the first item, that would be the expected behavior. Also for multiple selections in the members invite page, after inviting the members, the first item should not be highlighted. The only time the first item should be highlighted is when filtering the list.

**Comment by MonilBhavsar** [comment]
@fedirjh focussing on the first non selected item sounds good

**Comment by sourcecodedeveloper** [comment]
@MonilBhavsar The expected behavior is to focus on the selected item if the search filter is empty. The first item should only be selected if the search filter is not empty.

**Comment by MonilBhavsar** [comment]
Okay sounds good. Also confirmed it is the current behvaior in oldDot.
We should make it consistent across the lists. Looks like split bill selection behaves differently

**Comment by sourcecodedeveloper** [comment]
@fedirjh @MonilBhavsar 
In the country selection list, there is no default selection and the first item is not highlighted. The first item should not be highlighted if the search filter is empty. The behavior would be consistent with the timezone list where an item is selected by default.

[poc.webm](https://github.com/Expensify/App/assets/141340971/416443cb-86b2-4030-ae88-ac050a0a68a6)

To remove the highlight of the first item, currently the focusedIndex is being set to -1. But when focus is set to -1 the initiallyFocusedOptionKey also resets to empty string. Therefore, the initiallyFocusedOptionKey is not reliable to get the selected item. As an alternative as mentioned in the [proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128) , an index can be added to the selectedOptions array.

**Comment by sourcecodedeveloper** [comment]
@fedirjh 
For single selection
pt #3 When the text input is cleared, focus on the selected item (if available) if not then go to #1

**Comment by sourcecodedeveloper** [comment]
@fedirjh @MonilBhavsar Currently for the multiple selections list, the focusing is only handled if there are more than 1 sections e.g. the workspace invite list. The workspace members list only has 1 section and therefore focusing is not being handled. Focusing could be done on multiple selection list with just 1 section but the scope would increase. Also in regards to scope and what should be refactored in the current issue related to the regression, single and multiple selections list are implemented in BaseSelectionsList and BaseOptionsList. The members invite page is using BaseSelectionsList to handle multiple selections and regression for the  workspace currency list is also using BaseSelectionList to handle single selection. The BaseOptionsList can be refactored for consistency in another issue as the scope would fall outside of regression. 
I have modified [proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128) to handle multiple selections list.

To summarize the expected behavior for BaseSelectionsList
For single selection

1. On the initial render, focus the initial item only if there is a pre-selected item; otherwise, no item should be focused.
2. Focus on the first item when the text input changes.
3. When the text input is cleared, focus on the selected item (if available) if not then go to 1. 

For multiple selections list:

1. If there are no selected items, do not focus on any item.
2. If there are selected items and the text input changes, focus on the first non-selected item.

**Comment by jliexpensify** [comment]
@fedirjh now that Monil has answered your question, what is the next step? Should @abzokhattab still be assigned?

**Comment by jliexpensify** [comment]
Bump @fedirjh

**Comment by sourcecodedeveloper** [comment]
@fedirjh [proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128) is ready for review.

**Comment by sourcecodedeveloper** [comment]
@fedirjh I have covered the expected behavior in the [proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128). Just needs your approval and assignment.

**Comment by abzokhattab** [comment]
Hi @fedirjh, i have updated [my proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1793425403) as well

**Comment by sourcecodedeveloper** [comment]
@fedirjh When deciding on proposal please remember that I was first to cover expected behavior in [proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128) and assist in discussing expected behavior.

**Comment by MonilBhavsar** [comment]
Contributors have updated proposals. @fedirjh you want to give it a final look?

**Comment by sourcecodedeveloper** [comment]
@fedirjh Please review revised [proposal](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128) based on the expected behavior https://github.com/Expensify/App/issues/29080#issuecomment-1861756947.

**Comment by MonilBhavsar** [comment]
👍 
Waiting for review from @fedirjh!

**Comment by MonilBhavsar** [comment]
@fedirjh 👋

**Comment by jliexpensify** [comment]
He's around, I bumped him on Slack!

**Comment by sourcecodedeveloper** [comment]
@fedirjh 
> @abzokhattab @sourcecodedeveloper With your proposal , the focus is not removed when the input is cleared for a multiple select list

According to the requirements,
For multiple selections list:

1.     If there are no selected items, do not focus on any item.
2.     If there are selected items and the text input changes, focus on the first non-selected item.

The focus is completely removed if no items are selected, in the second video posted, an item is selected and because of that pt #2 applies i.e. if there is selected item and text input changes, focus on first non-selected item. 

for multi-select lists to test completely removing focus, please post video showing no items are selected and that would confirm if any items are still focused.

**Comment by sourcecodedeveloper** [comment]
@fedirjh 
> @abzokhattab @sourcecodedeveloper With your proposal , the focus is not removed when the input is cleared for a multiple select list

According to the requirements,
For multiple selections list:

1. If there are no selected items, do not focus on any item.
2. If there are selected items and the text input changes, focus on the first non-selected item.

The focus is completely removed if no items are selected, in the second video posted, an item is selected and because of that pt #2 applies i.e. if there is selected item and text input changes, focus on first non-selected item. 

for multi-select lists to test completely removing focus, please post video showing no items are selected and that would confirm if any items are still focused.


> I just reviewed and tested the proposed proposals, I think the simplest solution is https://github.com/Expensify/App/issues/29080#issuecomment-1797811942 from @tienifr , the code looks simple and the difference with other solutions is that it also completely remove focus if text input is cleared for multiple select lists as stated in https://github.com/Expensify/App/issues/29080#issuecomment-1847033636
> 
> I encountered this bug with all proposals, inside the invite members , when we type some text , there is no focused item :

The first video posted does not have any item selected and that is why pt # 1 applies. Please rerun test for second video but do not select an items and then focus should be completely removed. Please post video for multi-select with no item selected to confirm focus is being removed. Thanks

**Comment by sourcecodedeveloper** [comment]
@fedirjh Perhaps the initial review had some oversight, I have tested and the [proposed](https://github.com/Expensify/App/issues/29080#issuecomment-1831949128) solution completely removes focus for multiple select list

> @abzokhattab @sourcecodedeveloper With your proposal , the focus is not removed when the input is cleared for a multiple select list

For reference, I have attached video the focus is completely removed when the input is cleared for multiple select list. 

[poc.webm](https://github.com/Expensify/App/assets/141340971/ee82e5d9-8d8c-43b8-a684-2a84cf3d6be1)

[poc.webm](https://github.com/Expensify/App/assets/141340971/c35aea0f-3953-4176-9eb2-0378240dd08b)

**Comment by sourcecodedeveloper** [comment]
@fedirjh I noticed another thing
> I encountered this bug with all proposals, inside the invite members , when we type some text , there is no focused item :

That would expected behavior according to pt 1, if no item is selected do not focus on any item. In order to focus item, an item needs to be selected then pt 2 would apply.

For multiple selections list:

1. If there are no selected items, do not focus on any item.
2. If there are selected items and the text input changes, focus on the first non-selected item.

**Comment by sourcecodedeveloper** [comment]
@fedirjh I noticed another thing
> I encountered this bug with all proposals, inside the invite members , when we type some text , there is no focused item :

That would be expected behavior according to pt 1, if no item is selected do not focus on any item or completely remove focus. An item is focused only when an item is selected and then according to pt 2 first non-selected item is focused.

For multiple selections list:

1. If there are no selected items, do not focus on any item.
2. If there are selected items and the text input changes, focus on the first non-selected item.

@MonilBhavsar please confirm expected behavior for when the first non-selected item should be focused. From the requirements, according to 2, there should be focused item only when there are selected items and the focus would be on first non-selected item.

**Comment by sourcecodedeveloper** [comment]
> I encountered this bug with all proposals, inside the invite members , when we type some text , there is no focused item :
@fedirjh please confirm, that would be expected behavior, for multiple selection list, the first non-selected item is focused only when there are selected items according to pt 2.

**Comment by sourcecodedeveloper** [comment]
> I encountered this bug with all proposals, inside the invite members , when we type some text , there is no focused item :

@fedirjh That is what I have been trying to understand, that would expected behavior according to requirements.

**Comment by MonilBhavsar** [comment]
Proposals are being reviewed. @fedirjh needs to take a look at above responses

**Comment by jjcoffee** [comment]
Hey @fedirjh or @abzokhattab could you confirm if the new PR will also fix https://github.com/Expensify/App/issues/34300?

**Comment by abzokhattab** [comment]
Yes the new PR will fix the issue you are referring to @jjcoffee  as [this condition](https://github.com/Expensify/App/blob/0e2f487a5e0f28f5755cfe46d5c842036222a24e/src/components/SelectionList/BaseSelectionList.js#L373) in the `useEffect` will prevent the refocus from occurring.

**Comment by jliexpensify** [comment]
Wow this is an old issue, and also a regression? @MonilBhavsar can you check this summary?

- Bug Reporter: @dhanashree-sawant $50
- Contributor: @abzokhattab (No payment as it was a regression from another issue)
- C+: @fedirjh $500

**Comment by jliexpensify** [comment]
Hi @abzokhattab - thanks for surfacing this! Just clarifying: were you paid for the other job? If so, the 50% regression penalty should have been applied there.

My understanding is that if a regression occurs, then the original Contributor is assigned to the new bug to also resolve that. I had always thought that this counted as part of the Contributor "resolving" the regression, but perhaps my understanding is incorrect? I'll review the .MD.

**Comment by jliexpensify** [comment]
Hi @abzokhattab - just circling back: the .MD isn't super clear (it just has one line around regression payments), but I confirmed with a colleague:

- When a regression occurs, the new bug/job is assigned to the original Contributor and C+
- It's expected that both parties will resolve the regression
- Once the regression is fixed and deployed, after 7 days, 50% is deducted from the original job, that cause the regression
- There is no payment for the new job (i.e. regression), as this forms part of the Contributor and C+'s responsibility to fix the regression

For this one, you'd see a 50% deduction on your original job. @fedirjh is paid the full amount as he's a new C+ assigned to the PR.

**Comment by jliexpensify** [comment]
Ok, so I reviewed the entire set of comments - as I am not an Engineer, I may not be understanding fully, but here's what I think:

- It looks like the regression was initially pointed out [here](https://github.com/Expensify/App/issues/29080#issuecomment-1790107221)
- So this would put this bug (dated Oct 9th, 2023) as the original job.
- If this is correct, then this would mean your payment is reduced to $250 - correct?

A couple of clarifying questions from me:

1. The initial payment was [dated](https://github.com/Expensify/App/issues/29080#issuecomment-1790419360) November 2nd. My assumption is that this wasn't for the regression, as the regression was reported on the 2nd, and there's no way we could have merged and deployed to production on the same day. Is this correct?

2. Assuming the above is correct, was the 2nd proposal that you [submitted](https://github.com/Expensify/App/issues/29080#issuecomment-1793425403) on November 4th directly addressing the regression?

3. And then all subsequent proposals updates of the November 4th proposal?

**Comment by dhanashree-sawant** [comment]
Not to disturb the communication but just to inform, thanks @jliexpensify, I have accepted the offer on upwork.

**Comment by MonilBhavsar** [comment]
> https://github.com/Expensify/App/issues/29080#issuecomment-1965491387

@jliexpensify @abzokhattab also needs to be paid $500, as this issue was fixed by two PR's. One PR that fixed original bug and another PR that fixed the regression.

**Comment by jliexpensify** [comment]
Got it, thanks @MonilBhavsar and apologies @abzokhattab - having that context really helps. My assumption was that this was a bug that was a regression, and every proposal after was focused on the regression.

I've adjusted it to pay @abzokhattab $500

**Comment by jliexpensify** [comment]
@MonilBhavsar just checking - you don't mean $500 + $250 (regression penalty) right?

**Comment by MonilBhavsar** [comment]
No, since they made a follow up to fix the issue completely. We are fine to pay $500

**Comment by jliexpensify** [comment]
Paid and job closed!
